float64[] high_z